tardis/tests/tests_slow/report.py

```diff
@@ -0,0 +1,233 @@
+        self.plot_images = []
+        # Length of all three lists are same.
+
+        # `generated` stores the date and time of generation of report.
+        # It is set in `pytest_sessionfinish` using datetime.now()
+        self.generated = None
+
+        # File path of image of comparison plots for a single test.
+        self.image_path = None
+
+        # Make a dokuwiki connection if dokuwiki import was successful.
+        if dokuwiki_available:
+            try:
+                self.doku_conn = dokuwiki.DokuWiki(
+                    url=dokuwiki_details["url"],
+                    user=dokuwiki_details["username"],
+                    password=dokuwiki_details["password"])
+            except gaierror, dokuwiki.DokuWikiError:
+                print "Dokuwiki connection not established, report upload failed!"
+                self.doku_conn = None
+
+    def add_plot(self, path):
+        """
+        This method is used by any single test method to add the file path of
+        image of corresponding comparison plot.
+        """
+        self.image_path = path
+
+    def _appendrow(self, result, report):
+        """
+        This method is called indirectly by `pytest_runtest_logreport` according
+        to result of a particular test method. It adds a single member each in
+        the three lists of class - test_logs, error_logs, plot_images.
+
+        During execution, this method gets called repeatedly and it keeps on
+        populating the three lists with details per each test method.
+        """
+        duration = getattr(report, 'duration', 0.0)
+
+        test_class_name = report.nodeid.split('::')[1]
+        test_method_name = report.nodeid.split('::')[-1]
+
+        # append a tuple (name, result, duration)
+        self.test_logs.append((
+            "{0} :: {1}".format(test_class_name, test_method_name),
+            "**{0}**".format(result), "{0:.2f}".format(duration)
+        ))
+
+        # Add the error log if any, else append a null string.
+        if report.longrepr:
+            self.error_logs.append(
+                "<code>{0}</code>".format(report.longrepr)
+            )
+        else:
+            self.error_logs.append("")
+
+        # Add the image hyperlink (dokuwiki syntax) in list, and upload the
+        # image to dokuwiki, if corresponding test method had set image_path
+        # member using add_plot method, else a null string.
+        if self.image_path is not None:
+            image_dokulink = "plots:{0}_{1}".format(
+                tardis.__githash__[0:7], self.image_path.split('/')[-1])
+
+            self.doku_conn.medias.add(image_dokulink, self.image_path)
+            self.plot_images.append("{0}{1}|300x225{2} ".format(
+                "{{", image_dokulink, "}}")
+            )
+        else:
+            self.plot_images.append("")
+        # Reset image_path variable for next test method.
+        self.image_path = None
+
+    def _generate_report(self):
```

Comment:
I imagined `_generate_report` to return a string. This string would then be passed to `_save_report`.